packages/transaction-controller/src/TransactionController.ts

```diff
@@ -590,6 +758,48 @@ export class TransactionController extends BaseControllerV1<
     }
   }
 
+  #getEthQuery({
+    networkClientId,
+    chainId,
+  }: {
+    networkClientId?: NetworkClientId;
+    chainId?: Hex;
+  }): EthQuery {
+    // if multichain is disabled, use the global ethQuery
+    if (!this.enableMultichain) {
+      return this.ethQuery;
+    }
+    let networkClient:
+      | AutoManagedNetworkClient<NetworkClientConfiguration>
+      | undefined;
+
+    if (networkClientId) {
+      try {
+        networkClient = this.getNetworkClientById(networkClientId);
+      } catch (err) {
+        log('failed to get network client by networkClientId');
+      }
+    }
+
+    if (!networkClient && chainId) {
+      try {
+        networkClientId = this.findNetworkClientIdByChainId(chainId);
+        networkClient = this.getNetworkClientById(networkClientId);
+      } catch (err) {
+        log('failed to get network client by chainId');
+      }
+    }
+
+    if (networkClient) {
+      return new EthQuery(networkClient.provider);
+    }
+
+    // NOTE(JL): we're not ready to drop globally selected ethQuery yet.
```

Comment:
Is there a functional limitation in removing all legacy support and insisting on multi-chain?

It seems we've covered all the bases and the only cost is some minor additional client logic?

Reply:
see my comment [here](https://github.com/MetaMask/core/pull/3643#discussion_r1471335552) about breaking changes and the feature flag